lib/src/controls/form/number_box.dart

```diff
@@ -10,6 +9,12 @@ import 'package:math_expressions/math_expressions.dart';
 const kNumberBoxOverlayWidth = 60.0;
 const kNumberBoxOverlayHeight = 100.0;
 
+typedef FormatFunction = String? Function(num? number);
+
+abstract interface class Formatter {
+  String format(dynamic number);
+}
+
```

Comment:
These could be named `NumberBoxFormatCallback` and `NumberBoxFormatter`, respectivelly, to avoid confusion with other formatters

Reply:
The `Formatter` should be named `NumberBoxFormatter`.

But the `FormatFunction` should be named `NumberBoxFormatFunction`.
This method is not a formatted callback, but rather it is used to format numbers.

My changes to the precision formatting function can be divided into three parts:

1. Simple specification of precision length: This allows users to directly specify the desired precision of a number without using any formatting methods.

2. Incorporating an inline formatting method: This method remains the same as before and is still the most commonly used way to format numbers.

3. A more complex and general Formatter class: This class provides more flexibility in formatting numbers, including the ability to customize the formatting rules and use different formatting styles.

Users can choose from any of these three approaches to achieve the desired formatting.

Reply:
Sure!